cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -334,57 +389,3 @@ def _cache_path_as_rooted(self, cache_path: str) -> RootedPath:
             return self._project_subpath(cache_path)
         else:
             return self._output_dir.join_within_root(cache_path)
-
```

Comment:
This commit should have happened before the first one, i.e. before implementing the logic, that way we'd end up with an overall smaller change diff.

Reply:
Squashed first and last two commits into one commit, which implements a static method of `_ComponentResolver` class for generating PURLs. Everyone okay with that before I force push :fearful: ?

Reply:
just make sure unit tests are passing for each commit in the PR (like they are now).

Reply:
@slimreaper35 after thoroughly examining the latest revision what I meant by

> This commit should have happened before the first one, i.e. before implementing the logic, that way we'd end up with an overall smaller change diff.

was that:

- in commit 1 you convert `generate_purl_for_package` to a static instance method
- in commit 2 you provide the actual implementation for the method
- in commit 3 you take care of the unit tests

If you squash everything in a single commit that (from logic POV) adds too many changes in a single commit whereas we want gradual well-structured changes.

I apologize for the misunderstanding that may have originated from my comment

> just make sure unit tests are passing for each commit in the PR (like they are now).

as I only put ^this comment in not to forget about running unit tests after each patch without actually paying much attention to the details you posted before me leading to the confusion that I may have approved of squashing the 2 commits together. After all, it's better to just push a new revision right away and see if there are more comments rather than waiting for approval for future changes in the next revision.

Reply:
@slimreaper35 do you plan on taking care of the commits as suggested ^above?

Reply:
I will try